### DISCUSSION TITLE
not finding the content

### DISCUSSION THREAD

**Discussion by ObedVega** [rossjrw/pr-preview-action#70] [Q&A] [answered]
Hello i have to sites with this feature.

One works perfectly after compiling all the files into html.

The other is just a site with static html file, doesnt compile nothing and these are the settings:
name: Deploy PR previews
on:
  push:
    branches:
      - main
  pull_request:
    types:
      - opened
      - synchronize    
jobs:
  deploy-preview:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: Deploy PR Preview
        uses: rossjrw/pr-preview-action@v1.4.3
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          preview-branch: gh-pages
          source-dir: .
          umbrella-dir: pr-preview
          action: auto
          
         *When open the PR it provides a link with my preview but that preview is empty, does anyone knows how to fix it or maybe im putting the wrong source-dir?
         
         Any help is well appreaciated.
         Thanks.

**Comment by rossjrw** [comment]
(moved this from an Issue to a Discussion, as it's a request for help and not a bug report/feature request)

It'd be helpful to have a few more concrete details. Could you link the repository and the workflow logs, if possible? Your setup looks like it could be fine, but it's impossible to say for sure without knowing what you're working with.

**Reply by ObedVega** [reply]
im working in single html page where i need to show the pr previews but everythime it gives me the pr preview link, it looks empty

**Reply by rossjrw** [reply]
@ObedVega If you want specific help I'll need specific details, instead of vague descriptions of the problem. Could you link the repository and the workflow logs, if possible?

**Reply by ObedVega** [reply]
this is my repo with a PR: https://github.com/Teradata/developers.teradata.com/pull/28

This is the pr preview for static html files: https://developers.teradata.com/pr-preview/pr-28/

**Reply by rossjrw** [reply, marked as answer]
Looking at the workflow for your site's deployment, you're using an artifact-based deployment method rather than a branch-based deployment method.

https://github.com/Teradata/developers.teradata.com/blob/9df521276128764e34e4c3974dc04b6ad3ff8fe5/.github/workflows/static.yml

```yml
      - name: Upload artifact
        uses: actions/upload-pages-artifact@v1
        with:
          # Upload entire repository
          path: '.'
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v1
```

I suspect in your repository's Pages settings you'll find that it's been configured to deploy from 'GitHub Actions' rather than from a branch (which to be fair is named in a very misleading way):

![Screen Shot 2023-11-23 at 09 25 00 am](https://github.com/rossjrw/pr-preview-action/assets/29130152/e0024a7a-5fa0-4b3a-9210-4d953034b9ae)

Assuming that's the case, the very first instruction in PR Preview Action's README (https://github.com/rossjrw/pr-preview-action/blob/main/README.md) tells you not to do that, and to use 'Deploy from a branch'.

Therefore, if you want to use this action to make previews, you'll need to rebuild how the Teradata/developers.teradata.com repository deploys its pages site. It sounds like you've gotten this working once before, so I'll assume you know how to do that. However, if you don't want to do that (or if you can't because of e.g. org limitations), this action simply isn't compatible with that workflow.

There is an issue in this repository to add support for artifact-based workflows (https://github.com/rossjrw/pr-preview-action/issues/21) but is not being worked on currently.
